test/functorch/test_control_flow.py

```diff
@@ -259,6 +259,877 @@ def false_fn(x):
+
+                def true_fn(x):
+                    return x * x + 1
+
+                def false_fn(x):
+                    return x
+
+                x = cond(pred, true_fn, false_fn, (x,))
+
+            return x
+
+        x = torch.ones(4, requires_grad=True)
+        x_new = for_loop_test(x)
+        x_exp = for_loop_fake(x)
+
+        self.assertEqual(x_new, x_exp)
+
+        grad_out = torch.ones_like(x_new)
+        grads = torch.autograd.grad(x_new, (x,), grad_out)
+        expected_grads = torch.autograd.grad(x_exp, (x,), grad_out)
+        self.assertEqual(expected_grads, grads)
+
+        def f(x):
+            x_new = for_loop_test(x)
+            grad_out = torch.ones_like(x_new)
+            return torch.autograd.grad(x_new, (x,), grad_out)
+
+        # TODO: During compilation, the metadata of the true_fn has the
+        # requires_grad attribute set to False, while the false_fn has it
+        # set to true. Therefore, the tracing of the cond errors out.
+        # Similar to behavior to the test case test_cond_autograd_same_pytree_output
+        gm = make_fx(f, tracing_mode="symbolic")(x)
+        self.assertExpectedInline(
+            gm.code.strip(),
+            """\
+def forward(self, x_1):
+    mul = torch.ops.aten.mul.Tensor(x_1, x_1)
+    add = torch.ops.aten.add.Tensor(mul, 1);  mul = None
+    mul_1 = torch.ops.aten.mul.Tensor(add, add)
+    add_1 = torch.ops.aten.add.Tensor(mul_1, 1);  mul_1 = None
+    mul_2 = torch.ops.aten.mul.Tensor(add_1, add_1)
+    add_2 = torch.ops.aten.add.Tensor(mul_2, 1);  mul_2 = None
+    ones_like = torch.ops.aten.ones_like.default(add_2, pin_memory = False);  add_2 = None
+    mul_3 = torch.ops.aten.mul.Tensor(ones_like, add_1)
+    mul_4 = torch.ops.aten.mul.Tensor(ones_like, add_1);  ones_like = add_1 = None
+    add_3 = torch.ops.aten.add.Tensor(mul_4, mul_3);  mul_4 = mul_3 = None
+    mul_5 = torch.ops.aten.mul.Tensor(add_3, add)
+    mul_6 = torch.ops.aten.mul.Tensor(add_3, add);  add_3 = add = None
+    add_4 = torch.ops.aten.add.Tensor(mul_6, mul_5);  mul_6 = mul_5 = None
+    mul_7 = torch.ops.aten.mul.Tensor(add_4, x_1)
+    mul_8 = torch.ops.aten.mul.Tensor(add_4, x_1);  add_4 = x_1 = None
+    add_5 = torch.ops.aten.add.Tensor(mul_8, mul_7);  mul_8 = mul_7 = None
+    return (add_5,)""",  # noqa: B950
+        )
+
+    def test_cond_in_while(self):
+        def while_loop_fake(x):
+            def cond_fn(cnt, x):
+                return cnt < 3
+
+            def body_fn(cnt, x):
+                return (cnt + 1, x * x + 1)
+
+            cnt = torch.tensor(0)
+            cnt, x = _fake_while_loop(
+                cond_fn,
+                body_fn,
+                (
+                    cnt,
+                    x,
+                ),
+            )
+            return x, cnt
+
+        # TODO: Once the while loop does support autograd, replace the manual while
+        # loop with the higher_order_operator
+        def while_loop_test(x):
```

Comment:
Remove this test for now?

Reply:
I removed the while-loop test for the moment